lib/java/test-utils/src/main/java/org/enso/test/utils/ContextRule.java

```diff
@@ -0,0 +1,209 @@
+package org.enso.test.utils;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Engine;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.Value;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * Test rule that ensures that {@link Context} is initialized and discarded in tests. If used as
+ * {@link org.junit.ClassRule}, {@link Context} will be initialized just once for the whole test
+ * class. If used as {@link org.junit.Rule}, a new {@link Context} will be initialized for each test
+ * method.
+ *
+ * <p>This class simply delegates most of the methods either directly to {@link Context} or to
+ * {@link ContextUtils}.
+ *
+ * <p>Note that {@link ContextRule} cannot be used inside methods annotated with {@link
+ * org.junit.runners.Parameterized.Parameters}.
+ */
+public final class ContextRule implements TestRule {
+  private final Supplier<Context> contextSupplier;
+  private final ByteArrayOutputStream out;
+  private Context context;
+
+  private ContextRule(Supplier<Context> contextSupplier, ByteArrayOutputStream out) {
+    this.contextSupplier = contextSupplier;
+    this.out = out;
+  }
+
+  public static ContextRule createDefault() {
+    var out = new ByteArrayOutputStream();
+    Supplier<Context> supplier =
+        () -> {
+          return ContextUtils.defaultContextBuilder().out(out).err(out).build();
+        };
+    return new ContextRule(supplier, out);
+  }
+
+  public static ContextRule createCustom(Supplier<Context> contextSupplier) {
+    return new ContextRule(contextSupplier, null);
```

Comment:
`null` as `ByteArrayOutputStream`? Tests using this method don't get the _"print output when test fails only"_ behavior by default. That's unfortunate. We should do better when designing new API.

Reply:
Fixed in bdfaa6bfaaa3052d7baac7f46074f530ff2ed974

Reply:
On the latest revision f4a84e86e7f8a68662a8cfce040e3c9c70bfed00, `stdout` and `stderr` are never null, and always captured.